docs/source/examples/mg-mode-part-iii.ipynb

```diff
@@ -0,0 +1,1460 @@
+    "materials['Fuel'].add_element(elements['U'], 1, enrichment=3.)\n",
+    "\n",
+    "# Gadolinia bearing fuel\n",
+    "materials['Gad'] = openmc.Material(name='Gad')\n",
+    "materials['Gad'].set_density('g/cm3', 10.23)\n",
+    "materials['Gad'].add_element(elements['O'], 2)\n",
+    "materials['Gad'].add_element(elements['U'], 1, enrichment=3.)\n",
+    "materials['Gad'].add_element(elements['Gd'], .02)\n",
+    "\n",
+    "# Zircaloy\n",
+    "materials['Zirc2'] = openmc.Material(name='Zirc2')\n",
+    "materials['Zirc2'].set_density('g/cm3', 6.55)\n",
+    "materials['Zirc2'].add_element(elements['Zr'], 1)\n",
+    "\n",
+    "# Boiling Water\n",
+    "materials['Water'] = openmc.Material(name='Water')\n",
+    "materials['Water'].set_density('g/cm3', 0.6)\n",
+    "materials['Water'].add_element(elements['H'], 2)\n",
+    "materials['Water'].add_element(elements['O'], 1)\n",
+    "\n",
+    "# Boron Carbide for the Control Rods\n",
+    "materials['B4C'] = openmc.Material(name='B4C')\n",
+    "materials['B4C'].set_density('g/cm3', 0.7 * 2.52)\n",
+    "materials['B4C'].add_element(elements['B'], 4)\n",
+    "materials['B4C'].add_element(elements['C'], 1)\n",
+    "\n",
+    "# Steel \n",
+    "materials['Steel'] = openmc.Material(name='Steel')\n",
+    "materials['Steel'].set_density('g/cm3', 7.75)\n",
+    "materials['Steel'].add_element(elements['Fe'], 1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now create a Materials object that can be exported to an actual XML file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Instantiate a Materials object\n",
+    "materials_file = openmc.Materials(materials.values())\n",
+    "\n",
+    "# Export to \"materials.xml\"\n",
+    "materials_file.export_to_xml()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's move on to the geometry. The first step is to define some constants which will be used to set our dimensions and then we can start creating surfaces for the problem, the 8x8 lattice, the rods and the control blade.\n",
+    "\n",
+    "Before proceeding lets discuss some simplifications made to the problem geometry:\n",
+    "- To enable the use of an equal-width mesh for running the multi-group calculations, the intra-assembly gap was increased to the same size as the pitch of the 8x8 fuel lattice\n",
+    "- The can is neglected\n",
+    "- The pin-in-water geometry for the control blade is ignored and instead the blade is a solid block of B4C\n",
+    "- Rounded corners are ignored\n",
+    "- There is no cladding for the water rod"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Set constants for the problem and assembly dimensions\n",
```

Comment:
This entire code block as well as the following block to instantiate `half_spaces` could be greatly simplified by using `openmc.get_rectangular_prism(...)` to instantiate rectangular boxes. We should probably use this throughout our examples to simplify box creation.